dbms/CMakeLists.txt

```diff
@@ -72,8 +72,131 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
         endif ()
     endif ()
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wshadow")
-endif()
+    function (add_cxx_compile_options option)
+        add_compile_options(
+                "$<$<STREQUAL:$<TARGET_PROPERTY:LINKER_LANGUAGE>,CXX>:${option}>"
```

Comment:
I'm not an expert in CMake. I can guess that it means "add this option only if language is C++". Maybe it worth a comment.

Reply:
good idea, will add